tests/ttnn/unit_tests/operations/test_moreh_adamw.py

```diff
@@ -0,0 +1,226 @@
+        tt_exp_avg_sq_out,
+        tt_max_exp_avg_sq_out,
+        compute_kernel_config=compute_kernel_config,
+    )
+
+    assert tt_param_out.get_legacy_shape() == list(model.weight.shape)
+
+    param_result = tt_param_out.cpu().to(ttnn.ROW_MAJOR_LAYOUT).to_torch().to(torch.bfloat16)
+    exp_avg_result = tt_exp_avg_out.cpu().to(ttnn.ROW_MAJOR_LAYOUT).to_torch().to(torch.bfloat16)
+    exp_avg_sq_result = tt_exp_avg_sq_out.cpu().to(ttnn.ROW_MAJOR_LAYOUT).to_torch().to(torch.bfloat16)
+    if amsgrad:
+        max_exp_avg_sq_result = tt_max_exp_avg_sq_out.cpu().to(ttnn.ROW_MAJOR_LAYOUT).to_torch().to(torch.bfloat16)
+    else:
+        max_exp_avg_sq_result = None
+
+    whole_passing = True
+
+    rtol = atol = 0.1
+    pcc = 0.99
+    passing, out = comp_allclose_and_pcc(model.weight, param_result, pcc=pcc, rtol=rtol, atol=atol)
+    logger.debug(f"Out passing (param)={passing}")
+    logger.debug(f"Output pcc={out}")
+    whole_passing &= passing
+
+    passing, out = comp_allclose_and_pcc(cpu_exp_avg_result, exp_avg_result, pcc=pcc, rtol=rtol, atol=atol)
+    logger.debug(f"Out passing (exp_avg)={passing}")
+    logger.debug(f"Output pcc={out}")
+    whole_passing &= passing
+
+    passing, out = comp_allclose_and_pcc(cpu_exp_avg_sq_result, exp_avg_sq_result, pcc=pcc, rtol=rtol, atol=atol)
+    logger.debug(f"Out passing (exp_avg_sq)={passing}")
+    logger.debug(f"Output pcc={out}")
+    whole_passing &= passing
+
+    if amsgrad:
+        passing, out = comp_allclose_and_pcc(
+            cpu_max_exp_avg_sq_result, max_exp_avg_sq_result, pcc=pcc, rtol=rtol, atol=atol
+        )
+        logger.debug(f"Out passing (max_exp_avg_sq)={passing}")
+        logger.debug(f"Output pcc={out}")
+        whole_passing &= passing
+
+    assert whole_passing
+
+
+@pytest.mark.parametrize(
+    "shape",
+    [
+        [32, 32],  # single
+        [4, 3, 2, 6, 64, 64],  # multi tile
+    ],
+)
+@pytest.mark.parametrize("lr", [1e-2])
+@pytest.mark.parametrize("betas", [[0.5, 0.555]])
+@pytest.mark.parametrize("eps", [1e-08])
+@pytest.mark.parametrize("weight_decay", [0.3])
+@pytest.mark.parametrize("amsgrad", [True, False])
+@pytest.mark.parametrize("step", [8])
+def test_moreh_adamw(shape, lr, betas, eps, weight_decay, amsgrad, step, device):
+    torch.manual_seed(0)
+
+    run_moreh_adamw(shape, lr, betas, eps, weight_decay, amsgrad, step, device)
+
+
+@pytest.mark.parametrize(
+    "shape",
+    [[32, 32]],  # single
+)
+@pytest.mark.parametrize("lr", [1e-2])
+@pytest.mark.parametrize("betas", [[0.5, 0.555]])
+@pytest.mark.parametrize("eps", [1e-08])
+@pytest.mark.parametrize("weight_decay", [0.3])
+@pytest.mark.parametrize("amsgrad", [True, False])
+@pytest.mark.parametrize("step", [8])
+def test_moreh_adamw_callback(shape, lr, betas, eps, weight_decay, amsgrad, step, device, use_program_cache):
+    torch.manual_seed(0)
+    for _ in range(2):
+        run_moreh_adamw(shape, lr, betas, eps, weight_decay, amsgrad, step, device)
```

Comment:
```suggestion
        run_moreh_adamw(shape, lr, betas, eps, weight_decay, amsgrad, step, device)
        tt_dummy_tensor = ttnn.empty([1,1,32,32], ttnn.bfloat16, ttnn.TILE_LAYOUT, device)
```
Please add a dummy tensor as shown above. 
Originally, we didn't create a separate dummy tensor because the input and output tensor addresses were different in the second iteration even without a dummy. 
However, at some point, it was confirmed that the tensor created in the second iteration now has the same DRAM address.

Reply:
You mean that in the second iteration of every tests, allocated tensors on the device may share the same memory space, right? It's a bug in hardware side, isn't it?

Reply:
I don’t think it’s a bug; it’s just that the behavior has changed compared to before. 
If the tensors used in the first iteration are completely cleaned up before the start of the second iteration, it’s possible that tensors with the same memory address could be created in the second iteration.

Previously, even without a dummy tensor, the tensor addresses were different in the second iteration. However, now it seems like the same values are coming in. To address this, I plan to add a dummy tensor to ensure that the tensor addresses differ in the second iteration.

Reply:
Got it. I have added the dummy tensor

Reply:
```Python
        run_moreh_adamw(shape, lr, betas, eps, weight_decay, amsgrad, step, device)
        tt_dummy_tensor = ttnn.empty([1, 1, 32, 32], ttnn.bfloat16, ttnn.TILE_LAYOUT, device)
```
The dummy tensor should be placed under `run_moreh_adamw`. Otherwise, the dummy tensor could be included in both the first and second iterations, which may render it meaningless

Reply:
Updated